### PR TITLE
Compile Wildelands with latest Mac OS including M1 Arm Processor (Apple Silicion)

### LINKED ISSUE #5082: MacOS CI Failure
Since the `macos-latest` image was updated from  10.15.7 to   11.6 yesterday, all MacOS CI builds are now failing with a CXX compiler selftest error:
```

   Source:      ../
   Version:     1.1~git1[4374f79@HEAD]
   Destination: WidelandsRelease
   Type:        Debug
   macOS:       11.6
   Target:      10.7
   Compiler:    Apple clang version 12.0.5 (clang-1205.0.22.11)

-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/opt/ccache/libexec/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/local/opt/ccache/libexec/clang++
-- Check for working CXX compiler: /usr/local/opt/ccache/libexec/clang++ - broken
CMake Error at /usr/local/Cellar/cmake/3.21.3/share/cmake/Modules/CMakeTestCXXCompiler.cmake:62 (message):
-- Configuring incomplete, errors occurred!
  The C++ compiler
See also "/Users/runner/work/widelands/widelands/build_wl/CMakeFiles/CMakeOutput.log".

See also "/Users/runner/work/widelands/widelands/build_wl/CMakeFiles/CMakeError.log".
    "/usr/local/opt/ccache/libexec/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /Users/runner/work/widelands/widelands/build_wl/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/local/bin/ninja cmTC_80b5c && [1/2] Building CXX object CMakeFiles/cmTC_80b5c.dir/testCXXCompiler.cxx.o
    clang: warning: include path for libstdc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
    [2/2] Linking CXX executable cmTC_80b5c
    FAILED: cmTC_80b5c 
    : && /usr/local/opt/ccache/libexec/clang++ -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.7 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/cmTC_80b5c.dir/testCXXCompiler.cxx.o -o cmTC_80b5c   && :
    clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9 [-Wdeprecated]
    ld: library not found for -lstdc++
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:1 (project)


Error: Process completed with exit code 1.
```

### LINKED ISSUE #5103: Compiling widelands for Apple Silicon / M1 ARM
**Is your feature request related to a problem? Please describe.**

I  got a brand new M1 MacBook Pro and tried to compile widelands.

```
$ uname
Darwin <hostname> Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:01 PDT 2021; root:xnu-8019.41.5~1/RELEASE_ARM64_T6000 arm64
```
I switched from macports to brew. All the libs should be there but it now fails me at:
```
CMake Error at /opt/homebrew/Cellar/cmake/3.21.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Failed to find all ICU components (missing: ICU_LIBRARY) (found version
  "68.2")
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.21.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /opt/homebrew/Cellar/cmake/3.21.4/share/cmake/Modules/FindICU.cmake:335 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:146 (find_package)
```
**Describe the solution you'd like**

* I assume the OS / CPU detection from CMake needs to be adopted.
 * I will try to install some other sdl based games via brew or installer and maybe get some ideas there.

**Describe alternatives you've considered**

* As of the Rosetta Emulation Widelands runs emulated, so there is no hurry to get this done..
* Crosscompiling on M1 for Intel might be possible, but makes no sense.
* Crosscompiling vice versa, makes even less sense :-)

**Additional context**

This might actually help porting widelands to other ARM based Computers, e.g. the Rapberry

**Implementation steps**

- Create a new branch m1arm or such.
- Fix Issues with CMake detection 
- Fix potentially other Issue / Errors playing the game

### PR DESCRIPTION
**Pull request by klaus-halfmann**
@SirVer please check this may break you daily build.
Anyone else around with macOs Monterey?
How can we do a reasonable review here?

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -248,18 +248,20 @@ jobs:
...
+    runs-on: macos-11
     steps:
     - name: Installing dependencies
       run: |
         export ICU_ROOT="$(brew --prefix icu4c)" && \
         export PATH="/usr/local/opt/ccache/libexec:/usr/local/opt/gettext/bin:$PATH";
-        brew install boost ccache glew ninja sdl2 sdl2_image sdl2_mixer sdl2_ttf
+        brew install boost ccache glew ninja sdl2 sdl2_image sdl2_mixer sdl2_ttf dylibbundler
+        brewprefix="`brew --prefix`"
+        export CMAKE_PREFIX_PATH="${brewprefix}/opt/icu4c"
```
Comment (Noordfrees) [inline .github/workflows/build.yaml]: This line has no effect because changes to environment variables are not preserved between steps

To save this change:
```suggestion
        echo "CMAKE_PREFIX_PATH=${brewprefix}/opt/icu4c" >> $GITHUB_ENV
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -268,7 +270,7 @@ jobs:
       run: |
         mkdir build_wl
         cd build_wl/
-        ../utils/macos/build_app.sh ${{ matrix.compiler }} ${{ matrix.config }} ../
+        ../utils/macos/build_app.sh ${{ matrix.config }} ${{ matrix.compiler }}
```
Comment (Noordfrees) [inline .github/workflows/build.yaml]:
```suggestion
        ../utils/macos/build_app.sh "--${{ matrix.config }}" "--${{ matrix.compiler }}"
```
Then you can revert the changes to the matrix definition above

**Comment by klaus-halfmann** [comment]
Some Ubuntu testcase are failing, are they adresed somewhere?

**Comment by Noordfrees** [comment]
> Some Ubuntu testcase are failing, are they adresed somewhere?

Yes, I'm working on it: #5100
After #5081 and #5101 are both merged I can address the reproducible bionic/g++ testsuite failures.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -253,13 +253,15 @@ jobs:
         compiler:
           - clang
     name: MacOS ${{ matrix.config }} ${{ matrix.compiler }}
-    runs-on: macos-10.15  # Building with clang on MacOS 11 is currently broken
+    runs-on: macos-11
```
Comment (Noordfrees) [inline .github/workflows/build.yaml:256]: Please also update the name of the artifact in the second-to-last line in this file with the changed version

Comment (klaus-halfmann) [reply .github/workflows/build.yaml:256]: Done

**Comment by SirVer** [comment]
Yes, this might break nightly builds, but let's have it and I'll fix them in a followup if it creates any issues.

**Comment by klaus-halfmann** [comment]
@Noordfrees ca we have this now? my vacation will be over next week.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (Noordfrees) [reply .github/workflows/build.yaml]: You did not address this comment yet :)

Comment (klaus-halfmann) [reply .github/workflows/build.yaml]: Well it works without this change, so there is no need to change it?

Comment (Noordfrees) [reply .github/workflows/build.yaml]: If it's not needed, just remove these 2 lines. But there's no point adding commands that don't do anything

Comment (klaus-halfmann) [reply .github/workflows/build.yaml]: Done lets see how we are doing

**Comment by klaus-halfmann** [comment]
My local build is broken now :-( The binary is ok but it fails to locate the (existing?) data dir.
Waiting for the "official" build. Maybe merging with master changed something?

**Comment by klaus-halfmann** [comment]
OK its only broken inside the .dmg disk image (which is read only). So this is something else,
unrelated to this branch 😌.

**Review by Noordfrees — APPROVED** [review]
